bika/lims/content/instrument.py

```diff
@@ -654,13 +654,38 @@ def getCertificateExpireDate(self):
         return date
 
     def getWeeksToExpire(self):
-        """ Returns the amount of weeks untils it's certification expire
+        """
+        Returns the amount of weeks and days until the latest valid
+        certification for this instrument gets expired.
         """
         cert = self.getLatestValidCertification()
         if cert == None:
             return ''
-        date = cert.getValidTo().asdatetime().date();
-        return date - date.today()
+
+        date_to = cert.getValidTo()
+        if not date_to:
+            return 0, 0
+
+        expired = False
+        date_from = date.today()
+        date_to = date_to.asdatetime().date()
+        if date_from > date_to:
+            # Out of date, already expired
+            date_from = date_to
+            date_to = date.today()
+            expired = True
+
+        # Get the number of total days between the two dates
+        total_days = (date_to - date_from).days
+        days = total_days % 7
+        weeks = total_days / 7
+        if days > 0:
+            weeks = (total_days - days) / 7
+
+        days = -days if expired else days
+        weeks = -weeks if expired else weeks
+        return weeks, days
+
```

Comment:
I did this already here: https://github.com/bikalims/bika.lims/blob/master/bika/lims/content/instrument.py#L597